nixos/modules/services/security/authentik.nix

```diff
@@ -0,0 +1,306 @@
+        Authentik needs at least a secret key. To set a database password use AUTHENTIK_POSTGRESQL__PASSWORD:
+        ```
+        AUTHENTIK_SECRET_KEY=<secret>
+        AUTHENTIK_POSTGRESQL__PASSWORD=<pass>
+        ```
+      '';
+    };
+
+    secretFiles = mkOption {
+      type = types.attrsOf secret;
+      example = {
+        AUTHENTIK_SECRET_KEY = "/run/secrets/authentik_secret_key";
+        AUTHENTIK_POSTGRESQL__PASSWORD = "/run/secrets/authentik_postgress_passwd";
+      };
+      default = { };
+      description = ''
+        Attribute set containing paths to files to add to the environment of authentik.
+        The files are not added to the nix store, so they can be used to pass secrets to authentik.
+        Refer to the [documentation](https://docs.goauthentik.io/install-config/configuration/) for options.
+        Authentik needs at least a secret key. To set a database password use AUTHENTIK_POSTGRESQL__PASSWORD.
+      '';
+    };
+
+    outposts = {
+      ldap = mkOutpost "ldap";
+      radius = mkOutpost "radius";
+      proxy = mkOutpost "proxy";
+    };
+  };
+
+  config = mkIf cfg.enable {
+    assertions = [
+      {
+
+        assertion = cfg.database.createLocally -> cfg.database.name == cfg.database.user;
+        message = "The postgres module requires the database name and the database user name to be the same.";
+      }
+      {
+        assertion = cfg.environmentFile == null -> cfg.secretFiles ? "AUTHENTIK_SECRET_KEY";
+        message = ''
+          Authentik needs at least a secret key to run.
+          Use either the environmentFile or secretFiles.AUTHENTIK_SECRET_KEY to provide one.
+        '';
+      }
+    ];
+
+    services.authentik.environment = {
+      AUTHENTIK_LISTEN__HTTP = "${cfg.server.host}:${toString cfg.server.port}";
+      AUTHENTIK_POSTGRESQL__HOST = cfg.database.host;
+      AUTHENTIK_POSTGRESQL__PORT = toString cfg.database.port;
+      AUTHENTIK_POSTGRESQL__USER = cfg.database.user;
+      AUTHENTIK_POSTGRESQL__NAME = cfg.database.name;
+    };
+
+    systemd.services.authentik-server = {
+      description = "authentik server";
+      requires = after;
+      inherit after;
+      wantedBy = [ "multi-user.target" ];
+      environment = cfg.environment // cfg.server.environment;
+      serviceConfig = commonServiceConfig // {
+        ExecStart = startupScript [ "server" ] cfg.package "server";
+        EnvironmentFile = cfg.environmentFile;
+      };
+    };
+
+    systemd.services.authentik-worker = {
+      description = "authentik worker";
+      requires = after;
+      inherit after;
+      wantedBy = [ "multi-user.target" ];
+      environment = cfg.environment // cfg.worker.environment;
+      serviceConfig = commonServiceConfig // {
+        ExecStart = startupScript [ "worker" ] cfg.package "worker";
+        EnvironmentFile = cfg.environmentFile;
+      };
+    };
```

Comment:
With the default configuration, running the server and worker in the same netns will potentially cause crashes because the worker and server both listen on ports 9000 and 9300 for HTTP requests (healthchecks only for the worker) and metrics.